src/Reflection/Mixin/MixinMethodsClassReflectionExtension.php

```diff
@@ -29,6 +38,10 @@ private function findMethod(ClassReflection $classReflection, string $methodName
 	{
 		$mixinTypes = $classReflection->getResolvedMixinTypes();
 		foreach ($mixinTypes as $type) {
+			if (in_array($type->describe(VerbosityLevel::typeOnly()), $this->mixinExcludeClasses, true)) {
```

Comment:
Not sure if `$type->describe(VerbosityLevel::typeOnly())` is the best way to get the class names here.

Reply:
It's not, because it would skip generic types in `@mixin` like `Foo<T>`. The best way is to ask `$type instanceof TypeWithClassName` and then about `$type->getClassName()`.

Reply:
I thought `describe` would return `Foo<T>` if it is called on GenericObjectType.

What about the case when it's not an instance of `TypeWithClassName`?

Reply:
> I thought describe would return Foo<T> if it is called on GenericObjectType.

Yes, exactly, which is why your code wouldn't work if `Foo` should be excluded.

I realized you should probably use `TypeUtils::getDirectClassNames()` which is the easiest and also takes unions into account.